lib/internal/fs/promises.js

```diff
@@ -510,18 +510,15 @@ async function open(path, flags, mode) {
 async function read(handle, bufferOrOptions, offset, length, position) {
   let buffer = bufferOrOptions;
   if (!isArrayBufferView(buffer)) {
-    if (bufferOrOptions === undefined) {
-      bufferOrOptions = {};
-    }
-    if (bufferOrOptions.buffer) {
-      buffer = bufferOrOptions.buffer;
-      validateBuffer(buffer);
-    } else {
-      buffer = Buffer.alloc(16384);
-    }
-    offset = bufferOrOptions.offset || 0;
-    length = bufferOrOptions.length ?? buffer.byteLength;
-    position = bufferOrOptions.position ?? null;
+    bufferOrOptions ??= {};
```

Comment:
```suggestion
```

Reply:
Just in case, I don't inline this on purpose.
There is a room for future improvements, e.g. type hardening to avoid well-hidden bugs like `fh.read('/path/to/file')`.
Keeping this variable would reduce code churn in case of something like
```js
if (typeof bufferOrOptions !== 'object') {
  throw new ERR_INVALID_ARG_TYPE; // bye backwards compatibility
}
```
being added later.

Reply:
If we were to add a check in the future, I think in this case we would do
```js
if (bufferOrOptions != null) {
  validateObject(bufferOrOptions);
}
({/* destructuring */} = bufferOrOptions ?? ObjectCreate(null));
```
Use the syntax you find the more readable, I have a preference for the "inlined" version but both look good to me.

Reply:
Yep, but it depends on future code and "priority" of `isArrayBufferView` check in it.
Anyways, usually I also prefer to inline, but in this case I strongly believe that this line is much safer.